service/history/replication/stream_receiver.go

```diff
@@ -144,6 +144,14 @@ func (r *StreamReceiverImpl) Key() ClusterShardKeyPair {
 }
 
 func (r *StreamReceiverImpl) sendEventLoop() {
+	var panicErr error
+	defer func() {
+		if panicErr != nil {
+			r.MetricsHandler.Counter(metrics.ReplicationStreamPanic.GetMetricName()).Record(1)
+		}
+	}()
+	defer log.CapturePanic(r.logger, &panicErr)
```

Comment:
You can use [metrics.CapturePanic](https://github.com/temporalio/temporal/blob/main/common/metrics/panic.go#L41)